lib/auth/in_memory/checker.js

```diff
@@ -0,0 +1,176 @@
+    }
+    if (checkExists(data, 'User', 'keys', 'array', userObj)) {
+        userObj.keys.forEach(keyObj => {
+            // eslint-disable-next-line no-param-reassign
+            data.keys[keyObj.access] = incr(data.keys[keyObj.access]);
+        });
+    }
+}
+
+function checkAccount(data, accountObj) {
+    if (checkExists(data, 'Account', 'email', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.emails[accountObj.email] = incr(data.emails[accountObj.email]);
+    }
+    if (checkExists(data, 'Account', 'arn', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.arns[accountObj.arn] = incr(data.arns[accountObj.arn]);
+    }
+    if (checkExists(data, 'Account', 'canonicalID', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.canonicalIds[accountObj.canonicalID] =
+            incr(data.canonicalIds[accountObj.canonicalID]);
+    }
+
+    if (accountObj.users) {
+        if (checkType(data, 'Account', 'users', 'array', accountObj)) {
+            accountObj.users.forEach(userObj => checkUser(data, userObj));
+        }
+    }
+
+    if (accountObj.keys) {
+        if (checkType(data, 'Account', 'keys', 'array', accountObj)) {
+            accountObj.keys.forEach(keyObj => {
+                // eslint-disable-next-line no-param-reassign
+                data.keys[keyObj.access] = incr(data.keys[keyObj.access]);
+            });
+        }
+    }
+}
+
+function dumpCountError(property, obj) {
+    let count = 0;
+    Object.keys(obj).forEach(key => {
+        if (obj[key] > 1) {
+            log.error('property should be unique', {
+                property,
+                value: key,
+                count: obj[key],
+            });
+            ++count;
+        }
+    });
+    return count;
+}
+
+function dumpErrors(checkData) {
+    let nerr = dumpCountError('CanonicalID', checkData.canonicalIds);
+    nerr += dumpCountError('Email', checkData.emails);
+    nerr += dumpCountError('ARN', checkData.arns);
+    nerr += dumpCountError('AccessKey', checkData.keys);
+
+    if (nerr > 0) {
+        return true;
+    }
+
+    if (checkData.errors.length === 0) {
+        return false;
+    }
+
+    checkData.errors.forEach(msg => {
+        log.error(msg.txt, msg.obj);
+    });
+
+    log.fatal('invalid authentication config file (cannot start)');
+
+    return true;
+}
+
+/**
+ * @param {object} authdata - the authentication config file's data
+ * @return {boolean} true on erroneous data
+ *                   false on success
+ */
+export default function check(authdata) {
+    const checkData = {
+        errors: [],
+        emails: [],
+        arns: [],
+        canonicalIds: [],
+        keys: [],
+    };
+
+    if (authdata.accounts === undefined) {
+        checkData.errors.push({
+            txt: 'no "accounts" array defined in Auth config',
+        });
+        return dumpErrors(checkData);
```

Comment:
Sorry, I'm not sure I get the question, but my go-to answer is: There are no accounts to browse anyways, might as well return early.